### ISSUE TITLE
[Bug]:Chaos Test; When I kill proxy, I found dataCoord panic... 

### ISSUE THREAD

**Issue by WangErXiao** [milvus-io/milvus#16963]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.0.2
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

When I kill proxy node , I found datacoord crashed  and continue restarting;

[2022/05/13 02:57:58.223 +00:00] [FATAL] [logutil.go:102] [panic] [recover="\"Failed to create consumer by-prd-rootcoord-dml_4, error = All attempts results:\\nattempt #1:server error: ConsumerBusy: Exclusive consumer is already connected\\nattempt #2:server error: ConsumerBusy: Exclusive consumer is already connected\\nattempt #3:server error: ConsumerBusy: Exclusive consumer is already connected\\nattempt #4:server error: ConsumerBusy: Exclusive consumer is already connected\\nattempt #5:server error: ConsumerBusy: Exclusive consumer is already connected\\nattempt #6:server error: ConsumerBusy: Exclusive consumer is already connected\\nattempt #7:server error: ConsumerBusy: Exclusive consumer is already connected\\nattempt #8:server error: ConsumerBusy: Exclusive consumer is already connected\\nattempt #9:server error: ConsumerBusy: Exclusive consumer is already connected\\nattempt #10:server error: ConsumerBusy: Exclusive consumer is already connected\\nattempt #11:server error: ConsumerBusy: Exclusive consumer is already connected\\n\""] [stack="github.com/milvus-io/milvus/internal/util/logutil.LogPanic\n\t/go/src/github.com/milvus-io/milvus/internal/util/logutil/logutil.go:102\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:965\ngithub.com/milvus-io/milvus/internal/msgstream.(*mqMsgStream).AsConsumerWithPosition\n\t/go/src/github.com/milvus-io/milvus/internal/msgstream/mq_msgstream.go:165\ngithub.com/milvus-io/milvus/internal/msgstream.(*mqMsgStream).AsConsumer\n\t/go/src/github.com/milvus-io/milvus/internal/msgstream/mq_msgstream.go:131\ngithub.com/milvus-io/milvus/internal/datacoord.(*ChannelManager).unsubscribe\n\t/go/src/github.com/milvus-io/milvus/internal/datacoord/channel_manager.go:369\ngithub.com/milvus-io/milvus/internal/datacoord.(*ChannelManager).cleanUpAndDelete\n\t/go/src/github.com/milvus-io/milvus/internal/datacoord/channel_manager.go:607\ngithub.com/milvus-io/milvus/internal/datacoord.(*ChannelManager).processAck\n\t/go/src/github.com/milvus-io/milvus/internal/datacoord/channel_manager.go:578\ngithub.com/milvus-io/milvus/internal/datacoord.(*ChannelManager).watchChannelStatesLoop\n\t/go/src/github.com/milvus-io/milvus/internal/datacoord/channel_manager.go:661"]

### Expected Behavior

datacoord recover by self

**Comment by xiaofan-luan** [comment]
I've fixed the issue by adding retry logic.
The main issue is that pulsar failover happens slow,  we probably have to wait for a little bit longer for the failure recovery

**Comment by xiaofan-luan** [comment]
should be fixed by #17562
